### PR TITLE
TestPostContainersAttach: minor improvements

### LINKED ISSUE #36611: flaky test integration-cli/DockerSuite.TestPostContainersAttach()
Looking at recent PRs CI, there are a few failures like this:

```
11:03:32 
11:03:32 ----------------------------------------------------------------------
11:03:32 FAIL: docker_api_attach_test.go:98: DockerSuite.TestPostContainersAttach
11:03:32 
11:03:32 docker_api_attach_test.go:211:
11:03:32     c.Assert(actualStdout.Bytes(), checker.DeepEquals, []byte("hello\nsuccess"), check.Commentf("Attach didn't return the expected data from stdout"))
11:03:32 ... obtained []uint8 = []byte{0x73, 0x75, 0x63, 0x63, 0x65, 0x73, 0x73}
11:03:32 ... expected []uint8 = []byte{0x68, 0x65, 0x6c, 0x6c, 0x6f, 0xa, 0x73, 0x75, 0x63, 0x63, 0x65, 0x73, 0x73}
11:03:32 ... Attach didn't return the expected data from stdout
11:03:32 
```

in particular:
* https://github.com/moby/moby/pull/36606 on **z** ([logs](https://jenkins.dockerproject.org/job/Docker-PRs-s390x/8958/console))
* https://github.com/moby/moby/pull/36609 on **powerpc** ([logs](https://jenkins.dockerproject.org/job/Docker-PRs-powerpc/9025/console))
* https://github.com/moby/moby/pull/36594 on **z** ([logs](https://jenkins.dockerproject.org/job/Docker-PRs-s390x/8960/console))

### PR DESCRIPTION
**Pull request by kolyshkin**
The API part of this test (the last test case in TestPostContainerAttach) is flaky, i.e. it sometimes fails on powerpc or s390 like this:

```
> FAIL: docker_api_attach_test.go:98: DockerSuite.TestPostContainersAttach
> docker_api_attach_test.go:211:
>     c.Assert(actualStdout.Bytes(), checker.DeepEquals, []byte("hello\nsuccess"), check.Commentf("Attach didn't return the expected data from stdout"))
> ... obtained []uint8 = []byte{0x73, 0x75, 0x63, 0x63, 0x65, 0x73, 0x73}
> ... expected []uint8 = []byte{0x68, 0x65, 0x6c, 0x6c, 0x6f, 0xa, 0x73, 0x75, 0x63, 0x63, 0x65, 0x73, 0x73}
> ... Attach didn't return the expected data from stdout
```

Is seems that "success" is returned but "hello\n" is not (as if ContainerAttachOptions.Logs was false).

After a careful code audit I still can't figure out why it fails. The two major differences from the other cases are:

 - it uses API (ContainerAttach()) rather than sockRequestHijack();
 - it uses stdcopy.StdCopy() not io.ReadFull(),

Nevertheless it looks _almost_ legit, except for ignoring the error from StdCopy. So I don't know how to fix it. Still, there are some things that can be improved here, including:

1. Get the container's stderr as well, make sure it's empty.

2. Check that stdcopy.StdCopy() did not return an error.

3. Use strings, so the test output will be more readable.

4. Nitpicks; move/remove comments, simplify var names.

This should ultimately fix https://github.com/moby/moby/issues/36611

### CODE REVIEW

**Comment by thaJeztah** [comment]
@kolyshkin could this have any relation to https://github.com/moby/moby/pull/36517 ?

**Comment by kolyshkin** [comment]
OK, the amended test revealed the error is returned from StdRead (that was previously ignored):
```
docker_api_attach_test.go:212:
    c.Assert(err, checker.IsNil)
... value *net.OpError = &net.OpError{Op:"read", Net:"unix", Source:(*net.UnixAddr)(0xc423f66680), Addr:(*net.UnixAddr)(0xc423f666a0), Err:(*poll.TimeoutError)(0x28347a0)} ("read unix @->/go/src/github.com/docker/docker/bundles/test-integration/docker.sock: i/o timeout")
```

**Comment by kolyshkin** [comment]
> @kolyshkin could this have any relation to #36517 ?

It certainly might; let's see if reverting it will help

**Comment by kolyshkin** [comment]
OK, two rounds with commit 37983921c9 ("Ensure a hijacked connection implements CloseWrite whenever its underlying") reverted shows no failures (with my modified test). It's still not too many runs though.

Let's try without the revert now.

**Comment by kolyshkin** [comment]
Win CI timed out, experimental has some unrelated failures. I am currently out of ideas for this one.

**Comment by tonistiigi** [comment]
@kolyshkin FYI #36663

**Comment by kolyshkin** [comment]
> @kolyshkin FYI #36663

@tonistiigi thanks! Also, the intermittent nature of the test case failure is explained by whether there's something in the buffer; if it is empty (which in our CI sometimes happens on slower hardware) the `hijackedConnCloseWriter` is not performed.

**Comment by coolljt0725** [comment]
ping @thaJeztah  @tonistiigi  @vdemeester

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Review by boaz0 — APPROVED** [review]
LGTM :+1:

File: integration-cli/docker_api_attach_test.go [hunk 1]
```diff
@@ -205,10 +208,15 @@ func (s *DockerSuite) TestPostContainersAttach(c *check.C) {
 	_, err = resp.Conn.Write([]byte("success"))
 	c.Assert(err, checker.IsNil)
 
-	actualStdout := new(bytes.Buffer)
-	actualStderr := new(bytes.Buffer)
-	stdcopy.StdCopy(actualStdout, actualStderr, resp.Reader)
-	c.Assert(actualStdout.Bytes(), checker.DeepEquals, []byte("hello\nsuccess"), check.Commentf("Attach didn't return the expected data from stdout"))
+	var outBuf, errBuf bytes.Buffer
+	_, err = stdcopy.StdCopy(&outBuf, &errBuf, resp.Reader)
+	if errors.Cause(err).(net.Error).Timeout() {
```
[the remark this answers is not in the document]
Comment (kolyshkin) [reply integration-cli/docker_api_attach_test.go]: Good catch, thank you, I have updated the commit

**Review by thaJeztah — APPROVED** [review]
LGTM
